kotest-assertions/kotest-assertions-android/src/main/java/io/kotest/android/matchers/textview/matchers.kt

```diff
@@ -18,41 +18,47 @@ infix fun TextView.shouldNotHaveText(text: String) = this shouldNot haveText(tex
 fun haveText(text: String) = object : Matcher<TextView> {
     override fun test(value: TextView) = MatcherResult(
         text.contentEquals(value.text),
-        "TextView should have text $text but was $value.text",
+        "TextView should have text $text but was ${value.text}",
         "TextView should not have text $text, but had"
     )
-    
-}
 
-infix fun TextView.shouldHaveTextColorId(@ColorRes colorId: Int) = this.shouldHaveTextColorId(colorId, null)
-fun TextView.shouldHaveTextColorId(@ColorRes colorId: Int, theme: Theme? = null) = if (VERSION.SDK_INT >= VERSION_CODES.M) {
-    this.shouldHaveTextColor(resources.getColor(colorId, theme))
-} else {
-    this.shouldHaveTextColor(resources.getColor(colorId))
 }
 
-infix fun TextView.shouldNotHaveTextColorId(@ColorRes colorId: Int) = this.shouldNotHaveTextColorId(colorId, null)
-fun TextView.shouldNotHaveTextColorId(@ColorRes colorId: Int, theme: Theme? = null) = if (VERSION.SDK_INT >= VERSION_CODES.M) {
-    this.shouldNotHaveTextColor(resources.getColor(colorId, theme))
-} else {
-    this.shouldNotHaveTextColor(resources.getColor(colorId))
+fun haveTextColorId(@ColorRes colorRes: Int, theme: Theme? = null) = object: Matcher<TextView> {
+   override fun test(value: TextView): MatcherResult {
+      val currentColor = value.currentTextColor
+      val resourceColor = ResourcesCompat.getColor(value.resources, colorRes, theme)
+      val resourceName = value.resources.getResourceName(colorRes)
+
+      return MatcherResult(
+         resourceColor == currentColor,
+         "TextView should have color resource $resourceName(#$resourceColor) but was #$currentColor",
+         "TextView should not have color $resourceName(#$resourceColor), but had"
```

Comment:
Here we will have a proper resource name plus the colour of that resource, a bit more precise than the raw colour matcher.